networkx/algorithms/assortativity/correlation.py

```diff
@@ -134,6 +134,12 @@ def degree_pearson_correlation_coefficient(G, x="out", y="in", weight=None, node
     r : float
        Assortativity of graph by degree.
 
+    Raises
+    ------
+    NetworkXError
+        If the graph 'G' is empty.
+        If the graph 'G' has nodes with degrees less than two.
```

Comment:
This is not the condition that triggers the exception - nodes with degree less than 2 are fine (otherwise the `path_graph(4)` example in the docstring would fail, since the end nodes have degree 1!). IIUC, the corner case is related to graphs that have *only* a single self-loop edge.

Reply:
> IIUC, the corner case is related to graphs that have only a single self-loop edge.

we also get `ValueError: x and y must have length at least 2.` for 1 edge `DiGraph`s. And there probably might be some other cases too where the `len(x) < 2 or len(y) < 2` condition might be false.

```.py
import networkx as nx
G = nx.DiGraph([(1, 2)])
G1 = nx.Graph(G) 
print(nx.degree_pearson_correlation_coefficient(G1))
print(nx.degree_pearson_correlation_coefficient(G))

# output
"""
nan
/usr/local/lib/python3.10/dist-packages/scipy/stats/_stats_py.py:4781: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.
  warnings.warn(stats.ConstantInputWarning(msg))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[<ipython-input-14-73f922e088fa>](https://localhost:8080/#) in <cell line: 5>()
      3 G1 = nx.Graph(G)
      4 print(nx.degree_pearson_correlation_coefficient(G1))
----> 5 print(nx.degree_pearson_correlation_coefficient(G))
      6 xy = nx.node_degree_xy(G, x="out", y="in")
      7 print(xy)

3 frames
[/usr/local/lib/python3.10/dist-packages/scipy/stats/_stats_py.py](https://localhost:8080/#) in pearsonr(x, y, alternative, method)
   4766 
   4767     if n < 2:
-> 4768         raise ValueError('x and y must have length at least 2.')
   4769 
   4770     x = np.asarray(x)

ValueError: x and y must have length at least 2.
"""
```